improver_tests/utilities/test_mathematical_operations.py

```diff
@@ -494,5 +494,48 @@ def test_dimension_preservation(self):
         self.assertListEqual(result_coord_order, expected_coord_order)
 
 
+class Test_fast_linear_fit(IrisTest):
+    """Test the fast_linear_fit method"""
+
+    def setUp(self):
+        """Creates some random data to represent x and y."""
+        array_size = 25
+        self.x = np.random.random(array_size)
+        self.y = np.random.random(array_size)
+
+    def use_lstsq(self):
+        """Uses numpy's leastsquare algorithm to fit the data as a comparison"""
+        x = np.stack([self.x, np.ones(len(self.x))]).T
+        return np.linalg.lstsq(x, self.y, rcond=-1)[0]
+
+    def linear_fit(self, shape=(25,), axis=-1):
+        """Compares the output of fast_linear_fit with numpy's leastsquare algorithm."""
+        expected_out = self.use_lstsq()
+        x = self.x.reshape(shape)
+        y = self.y.reshape(shape)
+        result = np.array(fast_linear_fit(x, y, axis=axis))
+        self.assertArrayAlmostEqual(expected_out, result)
+
+    def test_basic_linear_fit(self):
+        """Tests fast_linear_fit with 1D data."""
+        self.linear_fit()
+
+    def test_linear_fit_with_2D(self):
+        """Tests fast_linear_fit with 2D data."""
+        self.linear_fit(shape=(5, 5), axis=(-2, -1))
+
+    def test_mismatching_shape(self):
+        """Tests fast_linear_fit with mismatching shapes."""
+        x = self.x.reshape(5, 5)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable name "x" doesn't conform to '[a-z_][a-z0-9_]{2,30}$' pattern](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5319684)